crates/sargon/src/system/sargon_os/sargon_os_delete_account.rs

```diff
@@ -0,0 +1,144 @@
+use crate::prelude::*;
+use radix_engine_interface::blueprints::account::{
+    AccountRemoveAuthorizedDepositorInput as ScryptoAccountRemoveAuthorizedDepositorInput,
+    AccountRemoveResourcePreferenceInput as ScryptoAccountRemoveResourcePreferenceInput,
+};
+
+// ==================
+// Delete Account (Public)
+// ==================
+impl SargonOS {
+    pub async fn create_delete_account_manifest(
+        &self,
+        account_address: AccountAddress,
+    ) -> Result<TransactionManifest> {
+        let network_id = account_address.network_id();
+        let resource_preferences = self
+            .get_account_resource_preferences(network_id, account_address)
+            .await?;
+
+        let authorized_depositors = self
+            .get_account_authorized_depositors(network_id, account_address)
+            .await?;
+
+        let manifest = TransactionManifest::delete_account(
+            &account_address,
+            resource_preferences,
+            authorized_depositors,
+        );
+
+        Ok(manifest)
+    }
+}
+
+// ==================
+// Delete Account (Internal)
+// ==================
+impl SargonOS {
+    async fn get_account_resource_preferences(
```

Comment:
I attempted creating a generic method for this and `get_account_authorized_depositors`, since they are very similar: they both send an `AccountPageRequest` to the GW client, expect a `PageResponse<T>`  and if there are more items to fetch, fetch them.

However, I wasn't able to make the async call (`gatewayClient.account_resource_preferences(request).await`) inside a closure. Didn't want to spend too much time but something to keep in mind.

Reply:
I can help with, so you need an async closure as an argument to a function?